wpsite-limit-posts.php

```diff
@@ -1,428 +1,447 @@
+    {
+        wc_doing_it_wrong(__FUNCTION__, __('Cheatin&#8217; huh?', 'wpsite-limit-posts'), $this->version);
+    }
+
+    /**
+     * Unserializing instances of this class is forbidden.
+     */
+    public function __wakeup()
+    {
+        wc_doing_it_wrong(__FUNCTION__, __('Cheatin&#8217; huh?', 'wpsite-limit-posts'), $this->version);
+    }
+
+    /**
+     * Main WPsite_Limit_Posts instance.
+     *
+     * Ensure only one instance is loaded or can be loaded.
+     *
+     * @return WPsite_Limit_Posts
+     */
+    public static function instance()
+    {
+
+        if (is_null(self::$_instance) && !(self::$_instance instanceof WPsite_Limit_Posts)) {
+            self::$_instance = new WPsite_Limit_Posts();
+            self::$_instance->hooks();
+        }
+
+        return self::$_instance;
+    }
+
+    /**
+     * WPsite_Limit_Posts constructor.
+     */
+    private function __construct()
+    {
+
+    }
+
+    /**
+     * Add hooks to begin.
+     * @return void
+     */
+    private function hooks()
+    {
+
+        add_action('plugins_loaded', array($this, 'load_plugin_textdomain'));
+        add_action('init', array($this, 'register_post_status'));
+        add_action('wp_insert_post_data', array($this, 'stop_publish_post'), '99', 2);
+
+        if (is_admin()) {
+
+            $plugin = plugin_basename(__FILE__);
+            add_filter("plugin_action_links_$plugin", array($this, 'plugin_links'));
+
+            add_action('admin_menu', array($this, 'register_pages'));
+            add_action('admin_notices', array($this, 'posts_notice'));
+
+            foreach (array('post', 'post-new') as $hook) {
+                add_action("admin_footer-{$hook}.php", array($this, 'extend_submitdiv_post_status'));
+            }
+        }
+    }
+
+    /**
+     * Load the plugin text domain for translation.
+     * @return void
+     */
+    public function load_plugin_textdomain()
+    {
+
+        $locale = apply_filters('plugin_locale', get_locale(), 'wpsite-limit-posts');
+
+        load_textdomain(
+            'wpsite-limit-posts',
+            WP_LANG_DIR . '/wpsite-limit-posts/wpsite-limit-posts-' . $locale . '.mo'
+        );
+
+        load_plugin_textdomain(
+            'wpsite-limit-posts',
+            false,
+            $this->plugin_dir() . '/languages/'
+        );
+    }
+
+    /**
+     * Hooks to 'plugin_action_links_' filter
+     *
+     * @since 1.0.0
+     */
+    public function plugin_links($links)
+    {
+
+        $settings_link = '<a href="options-general.php?page=' . self::$settings_page . '">Settings</a>';
+        array_unshift($links, $settings_link);
+
+        return $links;
+    }
+
+    /**
+     * Hooks to 'init' and resgisters new post status type
+     *
+     * @since 1.0.0
+     */
+    public function register_post_status()
+    {
+
+        register_post_status('limited', array(
+            'label' => esc_html__('Limited', 'wpsite-limit-posts'),
+            'public' => false,
```

Comment:
Why did you change this value to false from true??

Reply:
What actually happened is that in old version its true that limited post would get pushed to (Limited section of Posts) showing that it didn't post although its publicly visible without the preview of the Admin but in new version what happened is that it does get pushed to (Limited section of Posts) and also not publicly visible if Admin doesn't preview it.